File: ignasr-app/src/main/java/limdongjin/ignasr/repository/IgniteRepository.java
package limdongjin.ignasr.repository;

import reactor.core.publisher.Mono;

public interface IgniteRepository {
    public <K, V> Mono<Mono<K>> putAsync(String cacheName, K key, V value);
    public <K, V> K put(String cacheName, K key, V value);
    public <K, V> V get(String cacheName, K key);
    public <K, V> int size(String cacheName);
}

File: ignasr-app-kt/src/main/kotlin/limdongjin/ignasr/repository/IgniteRepository.kt
package limdongjin.ignasr.repository

import reactor.core.publisher.Mono

interface IgniteRepository {
    fun <K: Any, V: Any> putAsync(cacheName: String, key: K, value: V): Mono<Mono<K>?>
    fun <K: Any, V: Any> put(cacheName: String, key: K, value: V): K
    fun <K: Any, V: Any> get(cacheName: String, key: K): V
    fun <K: Any, V: Any> size(cacheName: String): Int
}